userspace/libscap/scap_savefile.c

```diff
@@ -137,29 +137,50 @@ int32_t scap_write_proc_fds(scap_t *handle, struct scap_threadinfo *tinfo, scap_
 	block_header bh;
 	uint32_t bt;
 	uint32_t totlen = MEMBER_SIZE(scap_threadinfo, tid);  // This includes the tid
+	uint32_t nfds = 0;
+	uint32_t idx = 0;
 	struct scap_fdinfo *fdi;
 	struct scap_fdinfo *tfdi;
 
 	//
-	// First pass of the table to calculate the length
+	// First pass of the table to calculate the size
+	//
+	HASH_ITER(hh, tinfo->fdlist, fdi, tfdi)

```

Comment:
another case where you could use HASH_COUNT, I think.

Reply:
done